contract/src/main.rs

```diff
@@ -1086,6 +1309,29 @@ pub extern "C" fn call() {
     )
     .unwrap_or_revert();
 
+    let holder_mode: u8 =
+        get_optional_named_arg_with_user_errors(ARG_HOLDER_MODE, NFTCoreError::InvalidHolderMode)
+            .unwrap_or_revert_with(NFTCoreError::InvalidHolderMode);
+
+    let whitelist_lock: u8 = get_optional_named_arg_with_user_errors(
+        ARG_WHITELIST_MODE,
+        NFTCoreError::InvalidWhitelistMode,
+    )
+    .unwrap_or(0u8);
+
+    let contract_white_list: Vec<ContractHash> = get_optional_named_arg_with_user_errors(
+        ARG_CONTRACT_WHITELIST,
+        NFTCoreError::InvalidContractWhitelist,
+    )
+    .unwrap_or_default();
+
+    let nft_metadata_kind: u8 = get_named_arg_with_user_errors(
+        ARG_NFT_METADATA_KIND,
+        NFTCoreError::MissingNFTMetadataKind,
+        NFTCoreError::InvalidNFTMetadataKind,
+    )
+    .unwrap_or_revert();
+
```

Comment:
comments on these new imputed values would be helpful

Reply:
Added comments